### PR TITLE
magit-branch-config-popup: new popup

### PR DESCRIPTION
**Pull request by npostavs**
As discussed in https://github.com/magit/magit/issues/2533#issuecomment-179128807 and https://github.com/magit/magit/issues/2533#issuecomment-182097038.

Still need to add a mention in the manual.

### CODE REVIEW

File: lisp/magit.el [hunk 1]
```diff
@@ -1315,6 +1329,14 @@ Non-interactively DIRECTORY is (re-)initialized unconditionally."
     (local-set-key (vector (magit-popup-event-key ev))
                    'magit-invoke-popup-action)))
 
+(if magit-branch-separate-config-popup
+    (magit-define-popup-action 'magit-push-popup
```
Comment (kyleam) [inline lisp/magit.el]: s/magit-push-popup/magit-branch-popup/ ?

Comment (kyleam) [inline lisp/magit.el]: Oh, never mind. I see that was suggested in the comment you linked to.
Shouldn't the config popup also be available from the branch popup?

File: lisp/magit.el [hunk 2]
```diff
@@ -279,6 +279,12 @@ prefer the former, then you should add branches such as \"master\",
   :group 'magit-commands
   :type '(repeat string))
 
+(defcustom magit-branch-separate-config-popup nil
+  "Whether to have a separate popup for branch config options."
```
Comment (kyleam) [inline lisp/magit.el]: Should this mention that this needs to be set before Magit is loaded?

File: lisp/magit.el (hunk 1 above)
Comment (npostavs) [inline lisp/magit.el]: Oh right, I was using a loop to put it in several other popups, but then I decided to remove it, and I forgot to put the branch popup back.

**Comment by npostavs** [comment]
Hmm, I added the a reference in the manual, but I'm not sure how to update the texi properly. When I just `ox-texinfo` there were a bunch of whitespace changes which I assume means I should be loading `ox-texinfo+`, but after loading I get `Lisp error: (void-function nil)` while trying to export to texi.

File: lisp/magit.el (hunk 2 above)
Comment (npostavs) [inline lisp/magit.el]: yeah, done now.

**Comment by tarsius** [comment]
I intend to soon give my "lesser packages" some attention and deal with issues such as this. For now just edit the org file and I will generate the texi myself before merge.

Please add an entry in the release notes. Other than that this is ready for merge right?

**Comment by npostavs** [comment]
> Please add an entry in the release notes.

Done.

> Other than that this is ready for merge right?

Yes, I think so.

**Comment by tarsius** [comment]
Unfortunately re-evaluating `magit.el` after making changes to it, would result in a very repetitive `magit-branch-popup`. And we don't have to force users to set this before loading `magit.el`. Please have a look at my not-quite-finished approach at https://github.com/magit/magit/compare/jb/separate-branch-config.

**Comment by npostavs** [comment]
> Please have a look at my not-quite-finished approach at https://github.com/magit/magit/compare/jb/separate-branch-config.

Oh neat, I didn't realize the popup values can use `lambda` like that.

**Comment by tarsius** [comment]
Well, it's quite a hack and I don't think I bothered to document it.

**Comment by tarsius** [comment]
Anyway, I'll try to clean up my version and merge it before I release v2.6 later this month.

**Pull request closed without merging**